content/blog/tyranny-of-nits/index.md

```diff
@@ -0,0 +1,208 @@
+3. Cut debate short, and actually decide things.
+
+Considering, designing and implementing controversial changes should be staged ([Niko's proposal is great!](https://smallcultfollowing.com/babysteps/blog/2023/09/18/stability-without-stressing-the-out/)).
+It doesn't really matter whether this is formal or informal (just make it clear to contributors).
+You need to answer the following questions:
+
+1. Should we do this?
+2. How do we do this?
+3. Are the details of this solution good enough?
+
+The first question is often the most important: every library and language has an infinite amount of things it *could* do, and the set of things it *maybe should* do is only smaller in a way that only a mathematician could love.
+Answering the question of "should we do this" is in my opinion the most important goal of things like [RFC processes](https://rust-lang.github.io/rfcs/).
+Build some consensus, come to a decision, and hopefully prioritize your work so that you don't have too many open "we should do this!" ideas floating around at once.
+
+The next stage is where all of the big architectural discussions should take place.
+Hash it out on paper, maybe even build some prototypes if needed to test some theories.
+Then come together and build something, even if your preferred design didn't win.
+
+Finally, it comes to evaluating the final implementation.
+In Rust, this tends to be the stabiliation process, but in most projects, this is just PR review.
+Here, you should be focusing on clear bugs and polish problems: compromise conflicts are already hashed out and future work has already been written up as you work.
+
+The key thing to learn here is shouldn't go backwards in that process without *very* good reason, especially when a lot of effort has been invested at previous stages: it will waste time, threaten projects and burn people out like a sparkler.
+Don't, by way of hypothetical example, pick a fight in on a stabilization report (stage 3), after years of implementation work, to argue about whether the feature should exist at all (a stage 1 concern).
+It doesn't matter if you think you're right (or even if you *are* right): this is fundamentally damaging to the project and to the team, and that will almost certainly outweigh the delta caused by a bad decision.
+
+## Setting the stakes: consequences and reversability
+
+The tricky part about this is knowing when to draw a close to the first two phases.
+When is an idea good enough to try out?
+When is a strategic set of compromises good enough to start implementing?
+
+When I think about this, I use two axes:
+
+1. **Impact:** how impactful is this? How many users will this impact? How central is this to what we're building? How visible is this change?
+2. **Reverseability:** how hard will it be to fix any mistakes that we make here? Can we just revert the change? Will there be things built on top of it?
```

Comment:
I think it's worth explicitly mentioning that both higher impact and lower reverseability = more work required until the project can move on